### PR TITLE
Initial rough estimate of wit interfaces

### PR DESCRIPTION
**Pull request by autodidaddict**
This is just an initial rough draft of wit interfaces corresponding to wasmcloud current interfaces. I'm quite sure none of this compiles and I don't know if these interfaces share types properly. Will be able to validate these once the latest changes are available in wit-bindgen.

### CODE REVIEW

File: wit/keyvalue.wit [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    get: func(key: string) -> result<option<string>, error>
+
+	
+	set: func(key: string, value: string) -> result<_, error>
+
+	
+	delete: func(key: key) -> result<_, error>
+
+	
+	exists: func(key: key) -> result<bool, error>
```
Comment (brooksmtownsend) [inline wit/keyvalue.wit:14]: is this weird tab/spaces?

Is there an official recommendation with 2/4 space indentation for wit?

**Review by brooksmtownsend — COMMENTED** [review]
General question, are these interfaces that we'd use or would we use the wasi-cloud http/kv interfaces? Is the plan to use these to transition our codegen from smithy to wit, and then swap to the wasi-cloud interfaces once they are through the proposal / implementation phase?

File: wit/keyvalue.wit (hunk 1 above)
Comment (autodidaddict) [reply wit/keyvalue.wit:14]: I have no idea on either count.

**Comment by autodidaddict** [comment]
> General question, are these interfaces that we'd use or would we use the wasi-cloud http/kv interfaces? Is the plan to use these to transition our codegen from smithy to wit, and then swap to the wasi-cloud interfaces once they are through the proposal / implementation phase?

This would be the phase 1 transition period as outlined in https://github.com/wasmCloud/wasmCloud/issues/336

**Review by brooksmtownsend — APPROVED** [review]

File: wit/keyvalue.wit (hunk 1 above)
Comment (thomastaylor312) [reply wit/keyvalue.wit:14]: Yeah the 2 empty lines in between each line is really throwing me for some reason 😆

File: wit/keyvalue.wit [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package wasmcloud:keyvalue@0.1.0
+
+/// An interface for reading and writing keys and their respective values
+interface readwrite {
+    get: func(key: string) -> result<option<string>, error>
```
Comment (thomastaylor312) [inline wit/keyvalue.wit:5]: I know the goal of phase 1 was to match the current contracts and this interface doesn't support all the set and list stuff we currently have. I am (obviously) perfectly ok with doing it this way, but just wanted to make sure that won't violate the spirit of what you're trying to do with phase 1

**Review by thomastaylor312 — COMMENTED** [review]
Just one minor comment.

I'd also like to see a basic README that explains what is in this directory for those who land here and that links them back to the RFC

**Review by ricochet — COMMENTED** [review]
This is looking good. I'd like a chance to finish reviewing before EOB today

**Comment by autodidaddict** [comment]
Have you had a chance to review this yet @ricochet ?

**Pull request merged**